openpype/tools/loader/model.py

```diff
@@ -272,15 +272,17 @@ def setData(self, index, value, role=QtCore.Qt.EditRole):
 
                 # update availability on active site when version changes
                 if self.sync_server.enabled and version_doc:
-                    repre_info = self.sync_server.get_repre_info_for_versions(
-                        project_name,
-                        [version_doc["_id"]],
-                        self.active_site,
-                        self.remote_site
+                    repres_info = list(
+                        self.sync_server.get_repre_info_for_versions(
+                            project_name,
+                            [version_doc["_id"]],
+                            self.active_site,
+                            self.remote_site
```

Comment:
Also, does this logic fail to update when switching from a version to a hero version (where the hero version is not the same version number as the one that was previously selected?)

Reply:
It's based on version id so ti should be fine.

Reply:
Except that it only triggers in the else statement just above. Right?

Reply:
Hmm you're right. But that's out of bounds of this PR.

Reply:
Created issue https://github.com/pypeclub/OpenPype/issues/3711